dev/tasks/java-jars/github.yml

```diff
@@ -22,8 +22,15 @@
 jobs:
 
   build-cpp-ubuntu:
-    name: Build C++ libraries Ubuntu
-    runs-on: ubuntu-latest
+    {% set arch = '${{ matrix.platform.arch }}' %}
+    name: Build C++ libraries Ubuntu {{ arch }}
+    runs-on: {{ '${{ matrix.platform.runs_on }}' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - { runs_on: ["ubuntu-latest"], arch: "x86_64"}
+          - { runs_on: ["self-hosted", "Linux", "arm64"], arch: "aarch_64" }
     steps:
       {{ macros.github_checkout_arrow()|indent }}
       {{ macros.github_install_archery()|indent }}
```

Comment:
I think we might have to define `ARCH` on the `env` to tag and use the underlying manylinux image for the correct architecture. Similar to what we do on the wheels here:
https://github.com/apache/arrow/blob/main/dev/tasks/python-wheels/github.linux.yml#L30-L36

Reply:
It seems the current failure on the java-jars for Linux ARM is due to it using the docker image for amd64:
https://github.com/ursacomputing/crossbow/actions/runs/5474239168/jobs/9968766885#step:7:1137